### ISSUE TITLE
Add fsharp support to Swashbuckle

### ISSUE THREAD

**Issue by jkone27** [domaindrivendev/Swashbuckle.AspNetCore#2300]
swashbuckle currently [doesnt work with F# functional frameworks on top of aspnetcore](https://github.com/giraffe-fsharp/Giraffe/issues/488#issuecomment-884217398) (using endpoint routing).

It would be awesome if some workaround existed to fix that issue,

F# webapis in [Falco](https://www.falcoframework.com/) or [Saturn](https://saturnframework.org/tutorials/how-to-start.html) (or even plain [Giraffe](https://giraffe.wiki/)) are very nice and tidy, but atm they cannot have automatic openapi generation on server side, making them not 100% production ready.

Giraffe is one of the most used frameworks, and Saturn is built on top of giraffe.

Falco is newer but also very interesting.

Unfortunately atm aspnetcore apis done with those 3 beautiful frameworks sitting on top of aspnetcore do not have a working swashbuckle option :(

**Comment by d4sein** [comment]
No updates on this?

**Comment by martincostello** [comment]
@captainsafia FYI, maybe something to consider for the new OpenAPI support.

**Comment by captainsafia** [comment]
I posted a response to this in the linked issue. See [here](https://github.com/giraffe-fsharp/Giraffe/issues/488#issuecomment-2091220245).

TL;DR: Frameworks should implement an `IApiDescriptionProvider` interface as part of their implementation to opt-in to OpenAPI generation with all tools.
